python/ql/lib/semmle/python/dataflow/new/BarrierGuards.qll

```diff
@@ -3,8 +3,44 @@
 private import python
 private import semmle.python.dataflow.new.DataFlow
 
+private predicate stringConstCompare(DataFlow::GuardNode g, ControlFlowNode node, boolean branch) {
+  exists(CompareNode cn | cn = g |
+    exists(StrConst str_const, Cmpop op |
+      op = any(Eq eq) and branch = true
+      or
+      op = any(NotEq ne) and branch = false
+    |
+      cn.operands(str_const.getAFlowNode(), op, node)
+      or
+      cn.operands(node, op, str_const.getAFlowNode())
+    )
+    or
+    exists(IterableNode str_const_iterable, Cmpop op |
+      op = any(In in_) and branch = true
+      or
+      op = any(NotIn ni) and branch = false
+    |
+      forall(ControlFlowNode elem | elem = str_const_iterable.getAnElement() |
+        elem.getNode() instanceof StrConst
+      ) and
+      cn.operands(node, op, str_const_iterable)
+    )
+  )
+}
+
 /** A validation of unknown node by comparing with a constant string value. */
-class StringConstCompare extends DataFlow::BarrierGuard, CompareNode {
+class StringConstCompareBarrier extends DataFlow::Node {
```

Comment:
I wondered for a second whether this should be called `StringConstCompareSanitizer`, but I think using `StringConstCompareBarrier` is fine :+1: (and consistent with the new name in Ruby at least)

Reply:
Sounds like this is good to merge, then.